File: MATTHEW/leetcode/AIcoding/4.GPT-4o/erlang_solutions/LCR 174-er-cha-sou-suo-shu-de-di-kda-jie-dian-lcof.erl
-module(solution).
-export([search_bst/2]).

-record(node, {val, left = null, right = null}).

search_bst(Root, Target) ->
    case Root of
        null -> null;
        #node{val = Val} when Val =:= Target -> Root;
        #node{val = Val} when Val > Target -> search_bst(Root#node.left, Target);
        #node{val = Val} -> search_bst(Root#node.right, Target)
    end.

File: MATTHEW/leetcode/AIcoding/5.Qwen-3/rust_solutions/LCR 174-er-cha-sou-suo-shu-de-di-kda-jie-dian-lcof.rs
impl Solution {

#[derive(Debug, PartialEq, Eq)]
pub struct TreeNode {
    pub val: i32,
    pub left: Option<Box<TreeNode>>,
    pub right: Option<Box<TreeNode>>,
}

impl TreeNode {
    #[inline]
    pub fn new(val: i32) -> Self {
        TreeNode {
            val,
            left: None,
            right: None,
        }
    }
}

pub fn search_bst(root: Option<Box<TreeNode>>, val: i32) -> Option<Box<TreeNode>> {
    let mut current = root;
    while let Some(node) = current {
        if node.val == val {
            return Some(node);
        } else if node.val < val {
            current = node.right;
        } else {
            current = node.left;
        }
    }
    None
}
}